### PR TITLE
Fixed for ST_COMPONENT_MODE with a proxy

### PR DESCRIPTION
**Pull request by brnaba-aws**
### Motivation
- Include http proxy for secure tunneling when aws-iot-device-client is built in st_component_mode
- Issue number: #449 

#### Change summary
Http proxy configuration is loaded even if DISABLE_MQTT is defined. This allow for secure tunneling (aws-iot-device-client in st_component_mode) to work with a proxy.

### Testing
 **Is your change tested? If not, please justify the reason.**  
 **Please list your testing steps and test results.** 
- Tested by building the aws-iot-device-client in st_component_mode, with a proxy. I was able to established secure tunneling connection from AWS IoT Console.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/config/Config.cpp [hunk 1]
```diff
@@ -277,13 +277,12 @@ bool PlainConfig::LoadFromCliArgs(const CliArgs &cliArgs)
         thingName = cliArgs.at(PlainConfig::CLI_THING_NAME).c_str();
     }
 
-    bool loadFeatureCliArgs = tunneling.LoadFromCliArgs(cliArgs) && logConfig.LoadFromCliArgs(cliArgs);
+    bool loadFeatureCliArgs = tunneling.LoadFromCliArgs(cliArgs) && logConfig.LoadFromCliArgs(cliArgs) && httpProxyConfig.LoadFromCliArgs(cliArgs);
```
Comment (HarshGandhi-AWS) [inline source/config/Config.cpp:280]: you will also have to validate the HTTP proxy config. Can we update it over here? https://github.com/awslabs/aws-iot-device-client/blob/main/source/config/Config.cpp#L2689-L2708

**Review by HarshGandhi-AWS — COMMENTED** [review]
Can we know why you or your team are using device client in `ST_COMPONENT_MODE` mode? The component mode was only added to support Greengrass Secure Tunneling component and was not meant to be used by the customer.

**Comment by Kriechi** [comment]
@HarshGandhi-AWS yes, its for a ST use case with a custom GGv2 component.

**Review by HarshGandhi-AWS — APPROVED** [review]
LGTM

**Comment by HarshGandhi-AWS** [comment]
Pushing the change on behalf of @brnaba-aws . I see some authentication failures while running the CI CD tests. Will create an action item to work on it separately unblocking this PR from being merged.

**Pull request merged**